Tools/cases_generator/generate_cases.py

```diff
@@ -108,7 +135,7 @@ def effect_str(effects: list[StackEffect]) -> str:
         pushed: str | None
         match thing:
             case parsing.InstDef():
-                if thing.kind != "op":
+                if thing.kind != "op" or (thing.kind != "inst" and self.instrs[thing.name].is_viable_uop()):
```

Comment:
Unless you're introducing a new kind, couldn't this be simplified like this?
```suggestion
                if thing.kind == "instr" or self.instrs[thing.name].is_viable_uop():
```
Then again, why not write this metadata for everything, both 'instr' and 'op'?